### ISSUE TITLE
JS Doc @param Object Bug

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#23221]
_From @fleck on April 6, 2018 15:7_

- VSCode Version: 1.22.0
- OS Version: MacOS 10.13.3

Steps to Reproduce:
Create the following file:
```
/**
 * Create a notification.
 * @param {string} options.heading Text for the header.
 * @param {string} [options.href] Where the primary button links to. The image will also be linked
 * to this href if an image is included.
 * @param {string} [options.imageUrl] Optional image url.
 * @param {string} [options.body] Optional text for the body.
 * @param {string} [options.buttonText] Text of the primary button.
 * @param {string} [options.secondButtonText] Text of the secondary button.
 * @param {string} [options.secondHref] Href of secondary button.
 * @param {string} [type] Add error styling by passing 'Error' for this parameter.
 * @param {object} options Object properties are interpolated into the error template.
 */
const Notification = (options, type = 'default') => {
};

export default Notification;
```

import the previous file and hover over Notification();
```
import Notification from './Notification';

Notification();
```

The signature of the function param 'object' is wrongly classified as any, but the descriptions below are correct.
<img width="571" alt="signature incorrect" src="https://user-images.githubusercontent.com/1706321/38426151-e01e2c56-3983-11e8-81c5-caec69219438.png">

Move options object description to the first position
```
/**
 * Create a notification.
 * @param {object} options Object properties are interpolated into the error template.
 * @param {string} options.heading Text for the header.
 * @param {string} [options.href] Where the primary button links to. The image will also be linked
 * to this href if an image is included.
 * @param {string} [options.imageUrl] Optional image url.
 * @param {string} [options.body] Optional text for the body.
 * @param {string} [options.buttonText] Text of the primary button.
 * @param {string} [options.secondButtonText] Text of the secondary button.
 * @param {string} [options.secondHref] Href of secondary button.
 * @param {string} [type] Add error styling by passing 'Error' for this parameter.
 */
const Notification = (options, type = 'default') => {
};

export default Notification;
```
and now hover over Notification(); and you'll see that the signature is correct, but the descriptions are now wrong.

<img width="539" alt="bad description" src="https://user-images.githubusercontent.com/1706321/38426446-ab3315c8-3984-11e8-913d-389be8e9361b.png">

Does this issue occur when all extensions are disabled?: Yes

_Copied from original issue: Microsoft/vscode#47333_

**Comment by mjbvz** [comment]
_From @vscodebot[bot] on April 6, 2018 15:7_

(Experimental duplicate detection)
Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
- [Intellisense when specifying object properties (#40181)](https://www.github.com/Microsoft/vscode/issues/40181)

**Comment by mjbvz** [comment]
Not sure if the first example (with the out of order param member declaration) is a valid jsdoc or not

In the second case, we lose the main `options` comment though, along with all the `@param` tags for the members

**Comment by sandersn** [comment]
@mjbvz 
1. No, at least according to usejsdoc and typescript. The second example is correct.
2. I'm not sure how inline object types flow through the language service to Code. On the parse tree, at least, all the comments are on their individual `@param` tags, so it's probably a matter of combining their descriptions correctly for display. @andy-ms might know more about this, or I can go looking to find out.

**Comment by neekibo** [comment]
I hope this is related: TS server wrongly nest `@prop` definitions. See Screenshot. `prop1` and `prop2` shoulb be on the same level as `form`. But TS thinks they are properties of `form`.
![code-jsdocs-error](https://user-images.githubusercontent.com/986167/41580780-5b10a680-739c-11e8-9c6d-dbf24c5cb24a.png)

VSCode Version 1.24.1 with default TS version (2.9, I think)

**Comment by sandersn** [comment]
@neekibo nope, it’s not related. Do you mind opening a new issue?

**Comment by sandersn** [comment]
I'm not going to fix (1) since it's not valid jsdoc and is very difficult to parse. @andy-ms fixed (2) with #23276, so I'm going to close this bug.
